### ISSUE TITLE
[tune] Parallel tune calls not autoscaling ray cluster from head node

### ISSUE THREAD

**Issue by ANarayan** [ray-project/ray#14544]
I am running a program on a ray cluster (deployed on kubernetes) that runs multiple `tune.run` calls in parallel. I am facing an issue where tune is unable to autoscale and resize the cluster to support new trial runs. Each `tune.run` has a max concurrency of 2 where each trial has a `resources_per_trial={ 'gpu' : 1, 'cpu' : 1}.` I am running 7 `tune.run` calls in parallel. In theory, I should have 14 requests of the following format: `{'GPU': 1.0, 'CPU': 1.0}`. This being said, as per the monitor output below, only 4 requests are made despite the parallelization.

I have attached [my cluster yaml file](https://github.com/ray-project/ray/files/6104341/kubs_cluster.txt) (uploaded as a .txt file) and below is a rough outline of my code:

```
@ray.remote(num_cpus=0, resources={f"node:{hostname}": 0.01})
def launch_experiment(exp):
       tune.run(exp)

def run_experiments():
     ray.get([launch_experiment.remote(exp) for exp in experiments])]
```

Here is the complete output of the autoscaling monitor:
```
======== Autoscaler status: 2021-03-08 13:22:12.352421 ========
Node status
---------------------------------------------------------------
Healthy:
 1 ray-legacy-head-node-type
Pending:
 (no pending nodes)
Recent failures:
 (no failures)

Resources
---------------------------------------------------------------

Usage:
 0.0/15.0 CPU
 0.00/30.127 GiB memory
 0.00/15.039 GiB object_store_memory

Demands:
 {'CPU_group_0_9b0b10e67b673746c9f58e8c73c1b700': 0.0101, 'CPU_group_9b0b10e67b673746c9f58e8c73c1b700': 0.0101}: 1+ pending tasks/actors
 {'GPU_group_f9fc676f3ecfe7f35df9be3e877df3b8': 0.0101, 'GPU_group_0_f9fc676f3ecfe7f35df9be3e877df3b8': 0.0101}: 1+ pending tasks/actors
 {'CPU_group_9a62e8f305eeafc55488144b3967ceb2': 0.0101, 'CPU_group_0_9a62e8f305eeafc55488144b3967ceb2': 0.0101}: 1+ pending tasks/actors
 {'CPU_group_0_221396a2c6cecafae83efd730d4246b7': 0.0101, 'CPU_group_221396a2c6cecafae83efd730d4246b7': 0.0101}: 1+ pending tasks/actors
 {'CPU_group_fc9c0bdc21ddb62de72ac2f3d02a01b4': 0.0101, 'CPU_group_0_fc9c0bdc21ddb62de72ac2f3d02a01b4': 0.0101}: 1+ pending tasks/actors
 {'CPU_group_0_d0ff0cbbf05d996591764cd2b7acfcaf': 0.0101, 'CPU_group_d0ff0cbbf05d996591764cd2b7acfcaf': 0.0101}: 1+ pending tasks/actors
 {'GPU_group_0_99264de56b31616257d0313b3d0444cb': 0.0101, 'GPU_group_99264de56b31616257d0313b3d0444cb': 0.0101}: 1+ pending tasks/actors
 {'GPU_group_0_0ddcc646838b7b7f4742380b34037445': 0.0101, 'GPU_group_0ddcc646838b7b7f4742380b34037445': 0.0101}: 1+ pending tasks/actors
 {'CPU_group_5ad6ce7ba6709f401b4f575e14e20da2': 0.0101, 'CPU_group_0_5ad6ce7ba6709f401b4f575e14e20da2': 0.0101}: 1+ pending tasks/actors
 {'GPU_group_5f3fa57db90d18c634224eaab72d1b63': 0.0101, 'GPU_group_0_5f3fa57db90d18c634224eaab72d1b63': 0.0101}: 1+ pending tasks/actors
 {'CPU_group_0_0fcd2aadbd162eba7f431ce12b0c82fb': 0.0101, 'CPU_group_0fcd2aadbd162eba7f431ce12b0c82fb': 0.0101}: 1+ pending tasks/actors
 {'GPU_group_516374d25d6ee83cfb3c876abd6a170b': 0.0101, 'GPU_group_0_516374d25d6ee83cfb3c876abd6a170b': 0.0101}: 1+ pending tasks/actors
 {'GPU_group_0_3dffd5acbabe0c74acbe317cb062060a': 0.0101, 'GPU_group_3dffd5acbabe0c74acbe317cb062060a': 0.0101}: 1+ pending tasks/actors
 {'CPU_group_4bee145e812b67628f8f6caea7a6bdea': 0.0101, 'CPU_group_0_4bee145e812b67628f8f6caea7a6bdea': 0.0101}: 1+ pending tasks/actors
 {'CPU_group_0501c8ef0e5c6ad159ef27b4bc4f00e7': 0.0101, 'CPU_group_0_0501c8ef0e5c6ad159ef27b4bc4f00e7': 0.0101}: 1+ pending tasks/actors
 {'CPU_group_ede8b29edc9a483ba8c1de143d7a811f': 0.0101, 'CPU_group_0_ede8b29edc9a483ba8c1de143d7a811f': 0.0101}: 1+ pending tasks/actors
 {'CPU_group_0_c685d71534a8ce3253e0caadf5999984': 0.0101, 'CPU_group_c685d71534a8ce3253e0caadf5999984': 0.0101}: 1+ pending tasks/actors
 {'GPU_group_0_4b54d032dbdd022580d1fe1096c62243': 0.0101, 'GPU_group_4b54d032dbdd022580d1fe1096c62243': 0.0101}: 1+ pending tasks/actors
 {'GPU_group_0_76a5cd0c4e6488ca458f1c256d1979b8': 0.0101, 'GPU_group_76a5cd0c4e6488ca458f1c256d1979b8': 0.0101}: 1+ pending tasks/actors
 {'CPU_group_0_9cbce0ef5b1b50c61ef6b349592c1f52': 0.0101, 'CPU_group_9cbce0ef5b1b50c61ef6b349592c1f52': 0.0101}: 1+ pending tasks/actors
 {'GPU_group_0_82cdc048c08251c291dd8764f07bbb62': 0.0101, 'GPU_group_82cdc048c08251c291dd8764f07bbb62': 0.0101}: 1+ pending tasks/actors
 {'CPU_group_543ab98420ac1607812a8da67552da4e': 0.0101, 'CPU_group_0_543ab98420ac1607812a8da67552da4e': 0.0101}: 1+ pending tasks/actors
 {'CPU_group_0_cc6dd0ff4969dd6e7f99ff1e1f64e6fa': 0.0101, 'CPU_group_cc6dd0ff4969dd6e7f99ff1e1f64e6fa': 0.0101}: 1+ pending tasks/actors
 {'CPU_group_d741f7d9181f550d7c978e1d52f1f218': 0.0101, 'CPU_group_0_d741f7d9181f550d7c978e1d52f1f218': 0.0101}: 1+ pending tasks/actors
 {'GPU_group_f57b2f2c723d720019af916d0c174adb': 0.0101, 'GPU_group_0_f57b2f2c723d720019af916d0c174adb': 0.0101}: 1+ pending tasks/actors
 {'CPU_group_0_75ba05bb01a1d9ce20ac521934f40858': 0.0101, 'CPU_group_75ba05bb01a1d9ce20ac521934f40858': 0.0101}: 1+ pending tasks/actors
 {'CPU_group_0_1da58f324de76d5ea4a12229c3d5b661': 0.0101, 'CPU_group_1da58f324de76d5ea4a12229c3d5b661': 0.0101}: 1+ pending tasks/actors
 {'CPU_group_8694ef37fc1b04c566c463e3d1eb5cfe': 0.0101, 'CPU_group_0_8694ef37fc1b04c566c463e3d1eb5cfe': 0.0101}: 1+ pending tasks/actors
 {'CPU': 1.0, 'GPU': 1.0} * 1 (PACK): 2+ pending placement groups
2021-03-08 13:22:17,519	WARNING resource_demand_scheduler.py:643 -- The autoscaler could not find a node type to satisfy the request: [{'GPU': 1.0, 'CPU': 1.0}, {'CPU': 1.0, 'GPU': 1.0}]. If this request is related to placement groups the resource request will resolve itself, otherwise please specify a node type with the necessary resource https://docs.ray.io/en/master/cluster/autoscaling.html#multiple-node-type-autoscaling.
2021-03-08 13:22:17,520	WARNING resource_demand_scheduler.py:643 -- The autoscaler could not find a node type to satisfy the request:
 [{'GPU': 1.0, 'CPU': 1.0}, {'CPU': 1.0, 'GPU': 1.0}, {'CPU_group_9b0b10e67b673746c9f58e8c73c1b700': 0.0101, 'CPU_group_0_9b0b10e67b673746c9f58e8c73c1b700': 0.0101},  
 {'GPU_group_f9fc676f3ecfe7f35df9be3e877df3b8': 0.0101, 'GPU_group_0_f9fc676f3ecfe7f35df9be3e877df3b8': 0.0101},   {'CPU_group_0_9a62e8f305eeafc55488144b3967ceb2': 0.0101, 'CPU_group_9a62e8f305eeafc55488144b3967ceb2': 0.0101},   {'CPU_group_0_221396a2c6cecafae83efd730d4246b7': 0.0101, 'CPU_group_221396a2c6cecafae83efd730d4246b7': 0.0101}, {'CPU_group_fc9c0bdc21ddb62de72ac2f3d02a01b4': 0.0101, 'CPU_group_0_fc9c0bdc21ddb62de72ac2f3d02a01b4': 0.0101}, 
  {'CPU_group_0_d0ff0cbbf05d996591764cd2b7acfcaf': 0.0101, 'CPU_group_d0ff0cbbf05d996591764cd2b7acfcaf': 0.0101},   {'GPU_group_99264de56b31616257d0313b3d0444cb': 0.0101, 'GPU_group_0_99264de56b31616257d0313b3d0444cb': 0.0101},  
 {'GPU_group_0_0ddcc646838b7b7f4742380b34037445': 0.0101, 'GPU_group_0ddcc646838b7b7f4742380b34037445': 0.0101}, {'CPU_group_5ad6ce7ba6709f401b4f575e14e20da2': 0.0101, 'CPU_group_0_5ad6ce7ba6709f401b4f575e14e20da2': 0.0101}, {'GPU_group_5f3fa57db90d18c634224eaab72d1b63': 0.0101, 'GPU_group_0_5f3fa57db90d18c634224eaab72d1b63': 0.0101}, {'CPU_group_0fcd2aadbd162eba7f431ce12b0c82fb': 0.0101, 'CPU_group_0_0fcd2aadbd162eba7f431ce12b0c82fb': 0.0101}, {'GPU_group_516374d25d6ee83cfb3c876abd6a170b': 0.0101, 'GPU_group_0_516374d25d6ee83cfb3c876abd6a170b': 0.0101}, {'GPU_group_3dffd5acbabe0c74acbe317cb062060a': 0.0101, 'GPU_group_0_3dffd5acbabe0c74acbe317cb062060a': 0.0101}, {'CPU_group_4bee145e812b67628f8f6caea7a6bdea': 0.0101, 'CPU_group_0_4bee145e812b67628f8f6caea7a6bdea': 0.0101}, {'CPU_group_0_0501c8ef0e5c6ad159ef27b4bc4f00e7': 0.0101, 'CPU_group_0501c8ef0e5c6ad159ef27b4bc4f00e7': 0.0101}, {'CPU_group_0_ede8b29edc9a483ba8c1de143d7a811f': 0.0101, 'CPU_group_ede8b29edc9a483ba8c1de143d7a811f': 0.0101}, {'CPU_group_c685d71534a8ce3253e0caadf5999984': 0.0101, 'CPU_group_0_c685d71534a8ce3253e0caadf5999984': 0.0101}, {'GPU_group_4b54d032dbdd022580d1fe1096c62243': 0.0101, 'GPU_group_0_4b54d032dbdd022580d1fe1096c62243': 0.0101}, {'GPU_group_76a5cd0c4e6488ca458f1c256d1979b8': 0.0101, 'GPU_group_0_76a5cd0c4e6488ca458f1c256d1979b8': 0.0101}, {'CPU_group_9cbce0ef5b1b50c61ef6b349592c1f52': 0.0101, 'CPU_group_0_9cbce0ef5b1b50c61ef6b349592c1f52': 0.0101}, {'GPU_group_0_82cdc048c08251c291dd8764f07bbb62': 0.0101, 'GPU_group_82cdc048c08251c291dd8764f07bbb62': 0.0101}, {'CPU_group_543ab98420ac1607812a8da67552da4e': 0.0101, 'CPU_group_0_543ab98420ac1607812a8da67552da4e': 0.0101}, {'CPU_group_0_cc6dd0ff4969dd6e7f99ff1e1f64e6fa': 0.0101, 'CPU_group_cc6dd0ff4969dd6e7f99ff1e1f64e6fa': 0.0101}, {'CPU_group_0_d741f7d9181f550d7c978e1d52f1f218': 0.0101, 'CPU_group_d741f7d9181f550d7c978e1d52f1f218': 0.0101}, {'GPU_group_f57b2f2c723d720019af916d0c174adb': 0.0101, 'GPU_group_0_f57b2f2c723d720019af916d0c174adb': 0.0101}, {'CPU_group_0_75ba05bb01a1d9ce20ac521934f40858': 0.0101, 'CPU_group_75ba05bb01a1d9ce20ac521934f40858': 0.0101}, {'CPU_group_1da58f324de76d5ea4a12229c3d5b661': 0.0101, 'CPU_group_0_1da58f324de76d5ea4a12229c3d5b661': 0.0101}, {'CPU_group_8694ef37fc1b04c566c463e3d1eb5cfe': 0.0101, 'CPU_group_0_8694ef37fc1b04c566c463e3d1eb5cfe': 0.0101}]. 
If this request is related to placement groups the resource request will resolve itself, otherwise please specify a node type with the necessary resource https://docs.ray.io/en/master/cluster/autoscaling.html#multiple-node-type-autoscaling.
```
cc @richardliaw

**Comment by richardliaw** [comment]
@AmeerHajAli @javi-redondo can you please take a look at this and provide some instructions as to how to fix this? The resource demands seem fine, but the node types cannot be detected.

**Comment by AmeerHajAli** [comment]
cc @DmitriGekhtman

**Comment by AmeerHajAli** [comment]
Please reopen if this does not solve your issue.

**Comment by richardliaw** [comment]
Is one possible workaround to specify the GPU resources manually in the node type definition?

**Comment by DmitriGekhtman** [comment]
Unfortunately, I screwed this up -- the autodetected resources override the manual ones:
https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/_private/kubernetes/config.py#L85

Will open an issue to remind myself to fix it soon.

**Comment by DmitriGekhtman** [comment]
https://github.com/ray-project/ray/issues/14553

**Comment by ANarayan** [comment]
> Unfortunately, I screwed this up -- the autodetected resources override the manual ones:
> https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/_private/kubernetes/config.py#L85
> 
> Will open an issue to remind myself to fix it soon.

So once this is fixed (#14553 ), would I be able to use the Ray Cluster Launcher? Or would i still need to use the operator

**Comment by DmitriGekhtman** [comment]
Yes -- if you use Ray master in your local environment after that is fixed to launch the cluster.

**Comment by ANarayan** [comment]
> Yes -- if you use Ray master in your local environment after that is fixed to launch the cluster.

Great. Also, I tried using the Ray Operator set up as suggested above and am still having the original issue with nodes not scaling up.

Here is the monitor output:
`======== Autoscaler status: 2021-03-08 18:41:51.838533 ========
Node status
---------------------------------------------------------------
Healthy:
 1 head-node
Pending:
 (no pending nodes)
Recent failures:
 (no failures)

Resources
---------------------------------------------------------------

Usage:
 0.0/12.0 CPU
 0.00/0.000 GiB memory
 0.00/0.000 GiB object_store_memory

Demands:
 {'GPU_group_0_e899f4053aa6b3027ebf14aaf4d760f2': 0.0101, 'GPU_group_e899f4053aa6b3027ebf14aaf4d760f2': 0.0101}: 1+ pending tasks/actors
 {'GPU_group_86a91f9b3d18a51f26cc9a6bbe193d1f': 0.0101, 'GPU_group_0_86a91f9b3d18a51f26cc9a6bbe193d1f': 0.0101}: 1+ pending tasks/actors
 {'CPU_group_e48f1f2423bc79ef0f3734c86668b247': 0.0101, 'CPU_group_0_e48f1f2423bc79ef0f3734c86668b247': 0.0101}: 1+ pending tasks/actors
 {'GPU_group_60f6ae68c09e5dbd73bbbc00e742990d': 0.0101, 'GPU_group_0_60f6ae68c09e5dbd73bbbc00e742990d': 0.0101}: 1+ pending tasks/actors
 {'GPU': 1.0, 'CPU': 1.0} * 1 (PACK): 2+ pending placement groups
example-cluster:2021-03-08 18:41:51,839	DEBUG legacy_info_string.py:24 -- Cluster status: 0 nodes
 - MostDelayedHeartbeats: {'10.92.1.3': 0.09535050392150879}
 - NodeIdleSeconds: Min=810 Mean=810 Max=810
 - ResourceUsage: 0.0/12.0 CPU, 0.0 GiB/0.0 GiB memory, 0.0 GiB/0.0 GiB object_store_memory
 - TimeSinceLastHeartbeat: Min=0 Mean=0 Max=0
Worker node types:
example-cluster:2021-03-08 18:41:56,921	DEBUG resource_demand_scheduler.py:153 -- Cluster resources: [{'node:10.92.1.3': 1.0, 'memory': 627.0, 'object_store_memory': 313.0, 'CPU': 12.0}]
example-cluster:2021-03-08 18:41:56,921	DEBUG resource_demand_scheduler.py:154 -- Node counts: defaultdict(<class 'int'>, {'head-node': 1})
example-cluster:2021-03-08 18:41:56,921	DEBUG resource_demand_scheduler.py:165 -- Placement group demands: [placement_group_id: "\350\231\364\005:\246\263\002~\277\024\252\364\327`\362"
name: "_tune__0cf593a9"
bundles {
  bundle_id {
    placement_group_id: "\350\231\364\005:\246\263\002~\277\024\252\364\327`\362"
  }
  unit_resources {
    key: "CPU"
    value: 1.0
  }
  unit_resources {
    key: "GPU"
    value: 1.0
  }
}
creator_job_id: "\002\000\000\000"
creator_actor_id: "\377\377\377\377\377\377\377\377\377\377\377\377\377\377\377\377"
creator_actor_dead: true
, placement_group_id: "\206\251\037\233=\030\245\037&\314\232k\276\031=\037"
name: "_tune__4211fc74"
bundles {
  bundle_id {
    placement_group_id: "\206\251\037\233=\030\245\037&\314\232k\276\031=\037"
  }
  unit_resources {
    key: "CPU"
    value: 1.0
  }
  unit_resources {
    key: "GPU"
    value: 1.0
  }
}
creator_job_id: "\002\000\000\000"
creator_actor_id: "\377\377\377\377\377\377\377\377\377\377\377\377\377\377\377\377"
creator_actor_dead: true
]
example-cluster:2021-03-08 18:41:56,922	WARNING resource_demand_scheduler.py:643 -- The autoscaler could not find a node type to satisfy the request: [{'GPU': 1.0, 'CPU': 1.0}, {'GPU': 1.0, 'CPU': 1.0}]. If this request is related to placement groups the resource request will resolve itself, otherwise please specify a node type with the necessary resource https://docs.ray.io/en/master/cluster/autoscaling.html#multiple-node-type-autoscaling.
example-cluster:2021-03-08 18:41:56,922	DEBUG resource_demand_scheduler.py:211 -- Resource demands: [{'GPU': 1.0, 'CPU': 1.0}, {'GPU': 1.0, 'CPU': 1.0}, {'GPU_group_86a91f9b3d18a51f26cc9a6bbe193d1f': 0.0101, 'GPU_group_0_86a91f9b3d18a51f26cc9a6bbe193d1f': 0.0101}, {'GPU_group_e899f4053aa6b3027ebf14aaf4d760f2': 0.0101, 'GPU_group_0_e899f4053aa6b3027ebf14aaf4d760f2': 0.0101}, {'CPU_group_e48f1f2423bc79ef0f3734c86668b247': 0.0101, 'CPU_group_0_e48f1f2423bc79ef0f3734c86668b247': 0.0101}, {'GPU_group_60f6ae68c09e5dbd73bbbc00e742990d': 0.0101, 'GPU_group_0_60f6ae68c09e5dbd73bbbc00e742990d': 0.0101}]
example-cluster:2021-03-08 18:41:56,922	DEBUG resource_demand_scheduler.py:212 -- Unfulfilled demands: [{'GPU': 1.0, 'CPU': 1.0}, {'GPU': 1.0, 'CPU': 1.0}, {'GPU_group_86a91f9b3d18a51f26cc9a6bbe193d1f': 0.0101, 'GPU_group_0_86a91f9b3d18a51f26cc9a6bbe193d1f': 0.0101}, {'GPU_group_e899f4053aa6b3027ebf14aaf4d760f2': 0.0101, 'GPU_group_0_e899f4053aa6b3027ebf14aaf4d760f2': 0.0101}, {'CPU_group_e48f1f2423bc79ef0f3734c86668b247': 0.0101, 'CPU_group_0_e48f1f2423bc79ef0f3734c86668b247': 0.0101}, {'GPU_group_60f6ae68c09e5dbd73bbbc00e742990d': 0.0101, 'GPU_group_0_60f6ae68c09e5dbd73bbbc00e742990d': 0.0101}]
example-cluster:2021-03-08 18:41:56,922	WARNING resource_demand_scheduler.py:643 -- The autoscaler could not find a node type to satisfy the request: [{'GPU': 1.0, 'CPU': 1.0}, {'GPU': 1.0, 'CPU': 1.0}, {'GPU_group_86a91f9b3d18a51f26cc9a6bbe193d1f': 0.0101, 'GPU_group_0_86a91f9b3d18a51f26cc9a6bbe193d1f': 0.0101}, {'GPU_group_e899f4053aa6b3027ebf14aaf4d760f2': 0.0101, 'GPU_group_0_e899f4053aa6b3027ebf14aaf4d760f2': 0.0101}, {'CPU_group_e48f1f2423bc79ef0f3734c86668b247': 0.0101, 'CPU_group_0_e48f1f2423bc79ef0f3734c86668b247': 0.0101}, {'GPU_group_60f6ae68c09e5dbd73bbbc00e742990d': 0.0101, 'GPU_group_0_60f6ae68c09e5dbd73bbbc00e742990d': 0.0101}]. If this request is related to placement groups the resource request will resolve itself, otherwise please specify a node type with the necessary resource https://docs.ray.io/en/master/cluster/autoscaling.html#multiple-node-type-autoscaling.`

**Comment by DmitriGekhtman** [comment]
Thanks -- will look into it

**Comment by richardliaw** [comment]
Hmm, it's not clear to me that this is a Kubernetes issue though?

One thing that would be good here is a bit more introspection into the set of 'available node types' on the cluster and their available resources (presumably this is an issue for @wuisawesome ?)

**Comment by ANarayan** [comment]
> > example-cluster:2021-03-08 18:41:56,922	WARNING resource_demand_scheduler.py:643 -- The autoscaler could not find a node type to satisfy the request: [{'GPU': 1.0, 'CPU': 1.0}, {'GPU': 1.0, 'CPU': 1.0}]. If this request is related to placement groups the resource request will resolve itself, otherwise please specify a node type with the necessary resource https://docs.ray.io/en/master/cluster/autoscaling.html#multiple-node-type-autoscaling.
> 
> Looks suspicious, @ANarayan what does your cluster config look like?

@wuisawesome couldn't attach it as a yaml file, but here is as a .txt file. 
[kubs_cluster.txt](https://github.com/ray-project/ray/files/6105610/kubs_cluster.txt). This is the config I used with the Ray Cluster Launcher.

**Comment by wuisawesome** [comment]
Looks like we aren't detecting the GPUs in the spec (@DmitriGekhtman  you're more likely to know why than me :p). 

As a workaround, can you use the multi node type config (like what the newly updated `example-full` looks like https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/kubernetes/example-full.yaml) 

Then set a field `resources: { "CPU": 1, "GPU": 1 }` directly in the `worker_node` field?

**Comment by ANarayan** [comment]
> @ANarayan
> Forgot to ask a basic question -- what version of Ray is being used to launch the cluster and what version of Ray is being run from the container images?

I am using 1.1.0 in the launch the cluster and the python 3.7 nightly version in the container images.

**Comment by DmitriGekhtman** [comment]
Ah, could you try 1.2.0 (or anything newer, like current master) to launch?

**Comment by DmitriGekhtman** [comment]
Also, would you mind running the following on one of the pods in the Ray cluster to check the commit:
`python -c "import ray; print(ray.__commit__)"`

**Comment by DmitriGekhtman** [comment]
^ yep

**Comment by ANarayan** [comment]
alright, so by setting the cluster launcher version I am seeing some scale up. However, the number of requests doesn't seem to match the number of requests I believe should be getting made. In the code below, each `tune.run` has a max concurrency of 2 where each trial has a resources_per_trial={ 'gpu' : 1, 'cpu' : 1}. I am running 5 tune.run calls in parallel. In theory, I should have 10 requests of the following format: {'GPU': 1.0, 'CPU': 1.0}. However, only 2 requests are made.

this is a rough outline of the code:
`@ray.remote(num_cpus=0, resources={f"node:{hostname}": 0.01})
def launch_experiment(exp):
       tune.run(exp)

def run_experiments():
     ray.get([launch_experiment.remote(exp) for exp in experiments])]`

**Comment by wuisawesome** [comment]
@richardliaw this is back in your territory now :p

**Comment by richardliaw** [comment]
Thanks! I'm gonna close this in light of a new issue to avoid clutter.

**Comment by VishDev12** [comment]
Sorry for tacking this on here, but I just wanted to add a note that I faced the same issue when using the 1.2.0 cluster launcher with Docker. And as noted above, using the nightly version of Ray to launch the cluster seems to fix it; the Ray cluster itself ran smoothly on 1.2.0.
